nncf/torch/sparsity/movement/algo.py

```diff
@@ -0,0 +1,480 @@
+                )
+            )
+            sparsified_module = target_model.get_containing_module(node_name)
+            self._sparsified_module_info.append(
+                SparseModuleInfo(node_name, sparsified_module, sparsifying_operation))
+
+        return insertion_commands
+
+    def create_weight_sparsifying_operation(self, target_module_node: NNCFNode, compression_lr_multiplier: float):
+        sparse_cfg = SparseConfig(SparseStructure.FINE)
+        node_name = target_module_node.node_name
+        for configs_per_scopes in self._sparse_configs_by_scopes:
+            target_scopes = configs_per_scopes.target_scopes
+            if matches_any(node_name, target_scopes):
+                sparse_cfg = configs_per_scopes.sparse_config
+                break
+
+        return MovementSparsifier(target_module_node, sparse_cfg=sparse_cfg, frozen=False,
+                                  compression_lr_multiplier=compression_lr_multiplier)
+
+    def _build_controller(self, model: NNCFNetwork) -> PTCompressionAlgorithmController:
+        return MovementSparsityController(model, self._sparsified_module_info, self.config)
+
+
+class PrunableOp:
+    def __init__(self, op_addr: OperationAddress, op_mod: Optional[torch.nn.Module]):
+        self.op_addr = op_addr
+        self.op_mod = op_mod
+
+
+@ADAPTIVE_COMPRESSION_CONTROLLERS.register('pt_movement_sparsity')
+class MovementSparsityController(BaseSparsityAlgoController):
+    def __init__(self, target_model: NNCFNetwork, sparsified_module_info: List[SparseModuleInfo],
+                 config: NNCFConfig):
+        super().__init__(target_model, sparsified_module_info)
+        algo_config = extract_algo_specific_config(config, 'movement_sparsity')
+        self._distributed = False
+        sparsify_operations = [m.operand for m in self.sparsified_module_info]
+        params = deepcopy(algo_config.get('params', {}))
+        scheduler_cls = SPARSITY_SCHEDULERS.get('threshold_polynomial_decay') # TODO: hard coded this scheduler name
+        self._scheduler = scheduler_cls(self, params)
+        self._loss = ImportanceLoss(sparsify_operations, self.scheduler)
+
+        #TODO: review - perhaps not the right place
+        self.config = config
+        self.prunableops_per_group = self._get_group_of_prunable_ops()  # This is planned to be deleted
+        # self.visualize_groups_of_prunables()
+        # self.create_structured_sparsity_context()
+        self.prunable_sparsified_module_info_groups = self._get_group_of_prunable_sparsified_module_info()
+
+        if self._scheduler.enable_structured_masking:
+            model_family = params.get('model_family', 'auto')
+            if model_family == 'auto':
+                model_family = detect_supported_model_family(self.model)
+            if model_family not in STRUCTURED_MASK_STRATEGY.registry_dict:
+                nncf_logger.warning('No supported model for structured masking. Disabled.')
```

Comment:
@yujiepan-work we need to overwrite `self._scheduler.enable_structured_masking` else it will error our since structurization will proceed and we don't have a strategy for model enters this loop.

Reply:
patched here [a4c4e27](https://github.com/openvinotoolkit/nncf/pull/1319/commits/a4c4e2783674108b5d0c49473a6771111b94dec5)